File: Python/fibonacci/src/fibonacci.py
# In The Name Of God
# ========================================
# [] File Name : fibonacci
#
# [] Creation Date : 17-04-2015
#
# [] Created By : Parham Alvani (parham.alvani@gmail.com)
# =======================================
__author__ = 'Parham Alvani'


class Fibonacci:
    """
    simple class for making fibonacci number with divide & conquer and
     dynamic programming methods.
    """

    def __init__(self):
        """
        build fibonacci object for you
        :return: nothing
        """
        self.list = list([0, 1])

    def get_fibonacci(self, index):
        """
        :param index: fibonacci sequence number
        :return: fibonacci at sequence number
        :raise: TypeError if index not a integer
        """
        if not isinstance(index, int):
            raise TypeError
        if self.list.__len__() - 1 >= index:
            return self.list[index]
        else:
            while self.list.__len__() - 1 < index:
                self.list.append(self.list[-1] + self.list[-2])
            return self.list[index]


fibonacci = Fibonacci()
var = input("Please enter fibonacci sequence number ")
var = int(var)
print(fibonacci.get_fibonacci(var))

# Test string operations
s = str("Hello world of %s" % "python")
print(s)
print(fibonacci.list)

File: go/fibonacci/fibonacci.go
/*
 * In The Name Of God
 * ========================================
 * [] File Name : fibonacci.go
 *
 * [] Creation Date : 28-05-2015
 *
 * [] Created By : Parham Alvani (parham.alvani@gmail.com)
 * =======================================
 */
/*
 * Copyright (c) 2015 Parham Alvani.
 */

package fibonacci

func Fibonacci(n int) int {
	if n == 0 || n == 1 {
		return 1
	}

	// nolint: gomnd
	return Fibonacci(n-1) + Fibonacci(n-2)
}
